### PR TITLE
Cleanup How-To Subheadings

### PR DESCRIPTION
**Pull request by atravitz**
This is a small change but I'm hoping it'll make the sidebar smaller and more legible. I just shortened the subheadings for clarity and changed the instances of "statepoint" to "state point" for consistency.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

**Comment by cbkerr** [comment]
I don't think it's a huge issue having a mix of "statepoint" and "state point" in the docs, but if we're going to pick a standard, I would argue for "statepoint" for **consistency with the code**.

As I was learning signac, I thought of statepoint as a word coined by signac, and this is exemplified in the code. Functions and arguments are defined as if statepoint is one word, for example:

* `Project.build_job_statepoint_index()` *not* `Project.build_job_state_point_index()`.
* `Project.get_statepoint()`  *not* `Project.get_state_point()`
* `Project.open_job(statepoint=None)` *not* `Project.open_job(state_point=None)`

I don't think we should change these function and argument names. Therefore, I think standardizing on "statepoint" in the docs is consistent with how users should internalize the concept to better learn function and argument names.


These code examples are all from signac-core. I couldn't find code examples in signac-flow that use either term.

File: docs/source/recipes.rst [hunk 1]
```diff
@@ -171,7 +171,7 @@ Executing this operation will store the output of the matlab script within the j
     Clarify that in principle the only Python needed is the definition of the bash command as a string returned from a decorated Python function.
 
 
-How to implement MPI-parallelized operations
+MPI-parallelized operations
```
Comment (cbkerr) [inline docs/source/recipes.rst:174]:
```suggestion
Running MPI-parallelized operations
```

File: docs/source/recipes.rst [hunk 2]
```diff
@@ -262,7 +262,7 @@ Finally, instead of modifying the operation implementation, you could use a cust
 
 Storing the above template in a file called ``templates/script.sh`` within your project root directory will prepend *every* operation command with ``mpiexec`` and so on.
 
-How to enforce the execution of a specific operation for debugging
+Forcing the execution of a specific operation for debugging
```
Comment (cbkerr) [inline docs/source/recipes.rst:265]:
```suggestion
Enforcing the execution of a specific operation for debugging
```

**Comment by csadorf** [comment]
@atravitz Is this ready for another iteration of reviews? 
@cbkerr Would you be ok with merging this PR and have the "statepoint" discussion separately? Maybe we place it on the next dev meeting agenda?

**Comment by cbkerr** [comment]
@atravitz started a [thread on the Signac Slack](https://signac.slack.com/archives/CPEHE14R5/p1598039186000700) (August 21, 2020) where @bdice and @vyasr stated that the intentional decision was made previously (years ago, offline) to standardize on "state point" in the docs.

This resolves my comment from August 19.

File: docs/source/recipes.rst (hunk 2 above)
Comment (bdice) [reply docs/source/recipes.rst:298]: "Forcing" is probably the expected term here. Resolving.

**Review by bdice — APPROVED** [review]
This is a helpful change! Thanks @atravitz. I'm going to merge this since it has 2 approvals.

**Pull request merged**